### ISSUE TITLE
Add per-epoch loss term tracking to HPO trials

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#94]
## Context

The current HPO scan body discards all per-batch loss values:

\`\`\`python
# optimisation/optimization_train_loop.py line 355-361
def scan_body(carry, batch_data):
    curr_params, curr_opt_state = carry
    new_params, new_opt_state, terms, total = train_step_trial(...)
    return (new_params, new_opt_state), None  # ← discards losses
\`\`\`

This makes it impossible to debug why certain hyperparameter combinations fail or produce poor NSE. Production training tracks all loss terms per epoch.

## Related Issues

- **Depends on**: #91 (train loop rewrite — uses \`make_scan_body()\` which already returns losses)
- **Parallel-safe** with: #92, #93

## Proposed Change

After #91, the rewritten train loop uses \`make_scan_body()\` from \`src/training/epoch.py\`, which returns \`(carry, (terms, total))\`. Use this to:

1. **Aggregate per-epoch losses** — mean across batches
2. **Log to Optuna trial attributes** — store epoch losses for post-hoc analysis
3. **Print periodically** — every N epochs, log loss breakdown to console

\`\`\`python
# After lax.scan in epoch loop:
(params, opt_state), (batch_terms, batch_totals) = lax.scan(
    scan_body, (params, opt_state), scan_inputs
)

# Aggregate
avg_total = jnp.mean(batch_totals)
avg_terms = {k: float(jnp.mean(v)) for k, v in batch_terms.items()}

# Log to Optuna (optional, every N epochs)
if (epoch + 1) % log_freq == 0:
    trial.set_user_attr(f"losses_epoch_{epoch}", avg_terms)
\`\`\`

## Acceptance Criteria

- [ ] Epoch-average loss terms are computed (not discarded)
- [ ] Loss breakdown logged to console periodically
- [ ] Optionally stored in Optuna trial \`user_attrs\` for post-hoc analysis
